packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1001,7 +1015,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this._connected = connected;
 
         if (changeOfState) {
-           this.replayPendingStates();
+            this.deltaManager.off("op", this.onOp);
```

Comment:
I feel like we are missing validation here - that we went through all local ops and applied them. That's pre-requisite for correctness. And maybe we unregister from delta manager (and clear pendingLocalState) and then here just assert that it's undefined?

Reply:
There's an assert for this in PendingStateManager.replayPendingStates()